app/models/commitment.rb

```diff
@@ -13,19 +16,32 @@ class Commitment < ApplicationRecord
   has_and_belongs_to_many :governance_types
   import_by governance_types: :name
   has_and_belongs_to_many :actions
+  import_by actions: :name
   has_and_belongs_to_many :threats
-  has_many :progress_documents
+  import_by threats: :name
   has_many :links
-  has_one_attached :spatial_data
+  import_by links: :url
+  has_many :progress_documents
+  has_one_attached :geospatial_file
+
+  belongs_to :criterium, optional: true
 
-  validates :spatial_data, 
+  accepts_nested_attributes_for :links, reject_if: ->(attributes){ attributes['url'].blank? }, allow_destroy: true
+  accepts_nested_attributes_for :progress_documents, reject_if: ->(attributes){ attributes['document'].blank? }, allow_destroy: true
```

Comment:
Will this destroy existing links if they aren't provided in the JSON sent to the update controller action? Just wondering how we handle deleting existing links and progress documents when updating commitments

Reply:
i've added  _destroy to the permitted params in the commitments controller. in Rails when destroying a nested model via another model (like in this case we'd be submitting commitment params, but wanting to destroy associated records at the same time), you submit the params for the existing record but with _destroy: true. 

on the front end the record is normally hidden and the _destroy param added. not sure that will be handled on the front end with survey js, we should talk about it in stand up. the backend needs the id of the record and the _destroy param, e.g.:

```
{
  commitment: {
    links_attributes: [
      {
        id: id of link record to be destroyed,
        _destroy: true
      }
    ]
  }
}
```